PYIC-1860: Send VCs to storage system

Wynndow opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Send VCs to storage system

### Why did it change

This adds the logic and permissions to start sending VCs to the CI
storage system in all envs other than prod.

We don't want this to actually block anything, so any unexpected errors
that might be thrown are caught and logged.

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-1860](https://govukverify.atlassian.net/browse/PYIC-1860)

Wynndow committed:
PYIC-1860: Start sending VC to storage system

This adds the logic and permissions to start sending VCs to the CI
storage system in all envs other than prod.

We don't want this to actually block anything, so any unexpected errors
that might be thrown are caught and logged.

kerrr commented on lib/src/main/java/uk/gov/di/ipv/core/library/helpers/LogHelper.java:
```diff
@@ -88,6 +92,22 @@ public static void logInfoMessageWithFieldAndValue(
+        LOGGER.log(logLevel, message);
+        LoggingUtils.removeKeys(logField1.getFieldName(), logField2.getFieldName(), logFieldValue3);
+    }
+
```
I think we should be using a structured message rather than this mechanism.

Wynndow replied:
Yeah, i thought that. But the structured message blows up if you give it a value of `null`. The place this is being used is triggered by two scenarios and it means the response may not always have all the fields. I could add a bunch of null checks, but it might leave some slightly weird values in the logs. I think this way null values just don't get included.

Wynndow committed:
PYIC-1860: Fix up sonarcloud code smells

kerrr commented on lib/src/main/java/uk/gov/di/ipv/core/library/service/CIStorageService.java:
```diff
@@ -0,0 +1,74 @@
+
+import static uk.gov.di.ipv.core.library.config.EnvironmentVariable.CI_STORAGE_PUT_LAMBDA_ARN;
+
+public class CIStorageService {
```
Super picky, but I think the convention is CiStorageService

Wynndow replied:
That makes my skin crawl though.
Will update and live with it.

blakeyp commented on lib/src/main/java/uk/gov/di/ipv/core/library/helpers/LogHelper.java:
```diff
@@ -88,6 +92,23 @@ public static void logInfoMessageWithFieldAndValue(
+        LoggingUtils.appendKey(logField2.getFieldName(), logFieldValue2);
+        LoggingUtils.appendKey(logField3.getFieldName(), logFieldValue3);
+        LOGGER.log(logLevel, message);
+        LoggingUtils.removeKeys(logField1.getFieldName(), logField2.getFieldName(), logFieldValue3);
```
Should this be `logField3.getFieldName()` ?

Wynndow committed:
PYIC-1860: Address PR comments

Renames the CiStorage service and logs errors as a map, rather than by
adding extra fields.
